src/Microsoft.AspNetCore.Sockets.Client.Http/WebSocketsTransport.cs

```diff
@@ -178,10 +178,29 @@ public WebSocketsTransport(HttpOptions httpOptions, ILoggerFactory loggerFactory
         {
             try
             {
+                var endOfMessage = true;
                 while (true)
                 {
-                    var memory = _application.Output.GetMemory();
+#if NETCOREAPP2_1
+                    // If there was a read that had a 'false' EndOfMessage then we should skip the 0 byte read since there is an in progress frame
+                    if (endOfMessage)
+                    {
+                        var result = await socket.ReceiveAsync(Memory<byte>.Empty, CancellationToken.None);
```

Comment:
Comment about why we're doing a `0`-byte read (to wait for data without allocating any buffers)

Reply:
Also, couldn't we do this with `Array.Empty<byte>` on all platforms? Still wouldn't allocate any buffers. I suppose we wouldn't be using the `struct`-returning overload though.

Reply:
I should probably add a comment saying that 0 byte reads don't seem to be supported for other platforms

Reply:
Ah, ok.